### PR TITLE
refactor(examples/with-typescript-graphql): use codegen `TypedDocumentNode` and GraphQL Yoga for better DX and smaller bundle size

### PR DESCRIPTION
**Pull request by charlypoly**
Improve the Next.js with TypeScript + GraphQL example:

- [x] use GraphQL Code Generator instead of `graphql-let`: more widespread tool and smaller bundle size (types only generation vs code generation)
- [x] use GraphQL Yoga instead of Apollo Server Micro: for lighter bundle size as [stated here](https://github.com/vercel/next.js/pull/36155)
- [x] introduces GraphQL Code Generator on the API side for Resolvers typing

### CODE REVIEW

File: examples/with-typescript-graphql/.babelrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "presets": ["next/babel"],
-  "plugins": ["graphql-let/babel"]
+  "plugins": []
 }
```
Comment (balazsorban44) [inline examples/with-typescript-graphql/.babelrc:4]: if no special plugins are needed anymore, I recommend removing `.babelrc` to leverage SWC for builds. :+1:

Comment (charlypoly) [reply examples/with-typescript-graphql/.babelrc:4]: @balazsorban44 will do, thanks!

File: examples/with-typescript-graphql/lib/graphql-operations.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
```
Comment (balazsorban44) [inline examples/with-typescript-graphql/lib/graphql-operations.ts:1]: Shouldn't generated files be ignored from version control?

File: examples/with-typescript-graphql/lib/resolvers-types.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
+import { GraphQLResolveInfo } from 'graphql';
```
Comment (balazsorban44) [inline examples/with-typescript-graphql/lib/resolvers-types.ts:1]: https://github.com/vercel/next.js/pull/36240/files#r859704535

File: examples/with-typescript-graphql/lib/graphql-operations.ts (hunk 2 above)
Comment (charlypoly) [reply examples/with-typescript-graphql/lib/graphql-operations.ts]: You're right, fixed in [f9ce8cb](https://github.com/vercel/next.js/pull/36240/commits/f9ce8cb28d05e5598204af98d60e4517a63d2ab7)

File: examples/with-typescript-graphql/lib/apollo.ts [hunk 4]
```diff
@@ -16,7 +17,13 @@ export type ResolverContext = {
 function createIsomorphLink(context: ResolverContext = {}) {
   if (typeof window === 'undefined') {
     const { SchemaLink } = require('@apollo/client/link/schema')
-    const { schema } = require('./schema')
+    const { readFileSync } = require('fs')
+    const { makeExecutableSchema } = require('@graphql-tools/schema')
+
+    const schema = makeExecutableSchema({
+      typeDefs: readFileSync('lib/schema.graphql', 'utf8'),
```
Comment (belgattitude) [inline examples/with-typescript-graphql/lib/apollo.ts:24]: worth to check this on a vercel deployment. readFileSync is highly dependent on cwd

Comment (belgattitude) [inline examples/with-typescript-graphql/lib/apollo.ts:24]: worth to check this on a vercel deployment. readFileSync is highly dependent on cwd

File: examples/with-typescript-graphql/pages/api/graphql.ts [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
+import { createServer } from '@graphql-yoga/node'
+import { readFileSync } from 'node:fs'
 
-const apolloServer = new ApolloServer({ schema })
+import resolvers from 'lib/resolvers'
 
-export const config = {
-  api: {
-    bodyParser: false,
+const typeDefs = readFileSync('lib/schema.graphql', 'utf8')
```
Comment (belgattitude) [inline examples/with-typescript-graphql/pages/api/graphql.ts:6]: worth to check this on a vercel deployment. readFileSync is highly dependent on cwd

Comment (belgattitude) [inline examples/with-typescript-graphql/pages/api/graphql.ts:6]: worth to check this on a vercel deployment. readFileSync is highly dependent on cwd

Comment (charlypoly) [reply examples/with-typescript-graphql/pages/api/graphql.ts]: @belgattitude Solved in 4f00042ee that brings edge functions support with no fs required

**Comment by charlypoly** [comment]
@belgattitude does it look to you? 👀

File: examples/with-typescript-graphql/package.json [hunk 6]
```diff
@@ -1,42 +1,39 @@
 {
   "private": true,
   "scripts": {
-    "codegen": "graphql-let",
+    "codegen": "graphql-codegen",
```
Comment (ijjk) [inline examples/with-typescript-graphql/package.json:4]: Hi, seems the `codegen` step is failing with: 

```sh
Something went wrong Failed to load schema for "./lib/resolvers-types.ts"
        Failed to load schema from pages/api/graphql.ts:

        Unable to load from file "/Users/ijjk/dev/vercel/next.js/examples/with-typescript-graphql/pages/api/graphql.ts": /Users/ijjk/dev/vercel/next.js/examples/with-typescript-graphql/pages/api/graphql.ts:1
import { createServer } from '@graphql-yoga/node'
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

Comment (charlypoly) [reply examples/with-typescript-graphql/package.json]: @ijjk, indeed, it was broken, thanks for the heads-up!

Fixed in 2c16df4192900781f5b3c864340b0296356aa219

**Review by ijjk — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by belgattitude** [comment]
@charlypoly I was off, but looks great. Haven't used yoga in prod yet, but it's much more lightweight on lambdas... A good idea to switch he example.

Thanks a lot

**Comment by charlypoly** [comment]
Thank you @belgattitude! 🙇🏼

**Pull request merged**